Makefile

```diff
@@ -168,6 +168,22 @@ dep-ensure: ##@dependencies Dep ensure and apply all patches
 dep-install: ##@dependencies Install vendoring tool
 	go get -u github.com/golang/dep/cmd/dep
 
+update-geth: ##@dependencies Update geth (use GETH_BRANCH to optionally set the geth branch name)
+ifdef GETH_BRANCH
+	@# escape slashes
+	@GETH_BRANCH=$(echo $GETH_BRANCH | sed 's@\/@\\\/@g')
+	@# Update go-ethereum contraint branch
+	@sed -i 'N;N;s@\(\[\[constraint]]\n  name = "github.com\/ethereum\/go-ethereum"\n  branch =\)\(.*\)@\1 '"\"${GETH_BRANCH}\""'@g' Gopkg.toml
```

Comment:
Wouldn't `dep ensure -update github.com/ethereum/go-ethereum@$(GETH_BRANCH)` work?

Reply:
I gave this a try, but it seemed to hint towards modifying Gopkg.toml instead. E.g.:

```
$ dep ensure -update github.com/ethereum/go-ethereum@release/1.8
dep: WARNING: redundant prune option "non-go" set for "github.com/karalabe/hid"
dep: WARNING: redundant prune option "non-go" set for "github.com/ethereum/go-ethereum"
Invalid arguments passed to ensure -update:

  ✗ version constraint release/1.8 passed for github.com/ethereum/go-ethereum, but -update follows constraints declared in Gopkg.toml, not CLI arguments

update arguments validation failed
```